lib/showcase_widget.dart

```diff
@@ -84,6 +89,18 @@ class ShowCaseWidgetState extends State<ShowCaseWidget> {
     });
   }
 
+  void _onStart() {
+    if (widget.onStart != null && activeWidgetId < ids.length) {
+      widget.onStart(activeWidgetId, ids[activeWidgetId]);
```

Comment:
as these `onStart` and `onComplete` functions are optional, there are chances that these might be null. So handle that case, prefer to use null-safe call to these function invocation.

e.g. `widget.onStart?.call(<args>)`